File: rei-do-chopp-api/ReiDoChopp.DataTransfer/Users/Requests/UserRegisterRequest.cs
using System.ComponentModel.DataAnnotations;

namespace ReiDoChopp.DataTransfer.Users.Requests
{
    public class UserRegisterRequest
    {
        [Required]
        [EmailAddress]
        public string Email { get; set; }

        [Required]
        public string FirstName { get; set; }

        [Required]
        public string LastName { get; set; }

        [Required]
        public string Password { get; set; }

        [Required]
        public string CheckPassword { get; set; }

        public int[] RoleIds { get; set; }
    }
}

File: rei-do-chopp-site/src/app/users/models/requests/user-register-request.ts
export class UserRegisterRequest {
  Email: string;
  FirstName: string;
  LastName: string;
  Password: string;
  CheckPassword: string;
  RoleIds;

  constructor(params: Partial<UserRegisterRequest> = {}) {
    this.Email = params.Email ?? '';
    this.FirstName = params.FirstName ?? '';
    this.LastName = params.LastName ?? '';
    this.Password = params.Password ?? '';
    this.CheckPassword = params.CheckPassword ?? '';
    this.RoleIds = params.RoleIds || [];
  }
}
